### PR TITLE
Added plugin version print to stack trace

### PR DESCRIPTION
**Pull request by TiBarification**
In case of what happen in #877, i created this branch, it depends on master.

This changes related to #876 and will add version next to plugin name.

### CODE REVIEW

File: core/logic/DebugReporter.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@
 #include <stdarg.h>
 #include "DebugReporter.h"
 #include "Logger.h"
-#include "sm_stringutil.h"
+#include "../sm_stringutil.h"
```
[the remark this answers is not in the document]
Comment (Headline) [reply core/logic/DebugReporter.cpp]: `#include <sm_stringutil.h>` and adding core to the logic/AMBuilder cxx includes will do work fine, too.

@asherkin anything wrong with that?

**Comment by asherkin** [comment]
`logic` should definitely not have `core` in its include path, that could very easily cause lots of trouble.

File: core/logic/DebugReporter.cpp [hunk 2]
```diff
@@ -103,14 +111,19 @@ void DebugReport::GenerateCodeError(IPluginContext *pContext, uint32_t code_addr
 	ke::SafeVsprintf(buffer, sizeof(buffer), message, ap);
 	va_end(ap);
 
-	const char *plname = pluginsys->FindPluginByContext(pContext->GetContext())->GetFilename();
+	IPlugin* plugin = pluginsys->FindPluginByContext(pContext->GetContext());
+	const char *plname = plugin->GetFilename();
+	const char *plversion = plugin->GetPublicInfo()->version;
 	const char *error = g_pSourcePawn2->GetErrorString(err);
-
+	if (!IS_STR_FILLED(plversion))
```
Comment (Headline) [inline core/logic/DebugReporter.cpp]: We're flailing around a lot to get this macro in. I don't think we want to add this into public/, so checking `plversion[0] == '\0'` by hand here is much preferred at this point.

**Comment by asherkin** [comment]
The discussion (or lack thereof) in #876 needs to be addressed.

File: core/logic/DebugReporter.cpp [hunk 3]
```diff
@@ -68,14 +69,21 @@ void DebugReport::GenerateErrorVA(IPluginContext *ctx, cell_t func_idx, int err,
...
 	ke::SafeVsprintf(buffer, sizeof(buffer), message, ap);
 
-	const char *plname = pluginsys->FindPluginByContext(ctx->GetContext())->GetFilename();
+	IPlugin* plugin = pluginsys->FindPluginByContext(ctx->GetContext());
+	const char *plname = plugin->GetFilename();
+	const char *plversion = plugin->GetPublicInfo()->version;
+	
+	if (!IS_STR_FILLED(plversion))
+	{
+		plversion = "Not specified";
```
Comment (stickz) [inline core/logic/DebugReporter.cpp]: Should the version section be removed from error log, if there's none specified?

**Pull request closed without merging**